includes/wc-payment-api/class-wc-payments-api-client.php

```diff
@@ -318,6 +332,16 @@ public function update_intention(
 			$request['setup_future_usage'] = 'off_session';
 		}
 
+		if ( $this->is_fraud_prevention_enabled_for_store() ) {
+			$request['metadata']['fraud_prevention_data_available'] = true;
+
+			$request['metadata']['fraud_prevention_data'] = [
+				'payment_intent_id_hash' => $this->hash_data_for_fraud_prevention( $intention_id ),
+				'payment_method_hash'    => '' !== $selected_upe_payment_type ? $this->hash_data_for_fraud_prevention( $selected_upe_payment_type ) : null,
```

Comment:
I believe we have to hash e.g. 'card:<card-number>' here. It could be also done on the server side, where we could extract payment details from the transaction object. This pushes us toward having the hashing partially on server side.

Let;s keep this approach and when we address the comments, we'll follow up on this.

Reply:
Actually, related to the comment above about server- vs client-side, I think we can calculate this block on the server-side if needed. Front-end should only provide us with data that are not reliable to retrieve otherwise, e.g. IP, User-Agent, etc.

Reply:
That makes sense. I'll remove this block